### DISCUSSION TITLE
Google cannot index my Next.js app hosted on Vercel

### DISCUSSION THREAD

**Discussion by duckception** [vercel/community#6796] [Help]
### Summary

I'm hosting my Next.js app on Vercel, which cannot be indexed by Google for some reason. I get the "Page cannot be indexed: Server error (5xx)" error in the Google console even though the app is accessible normally through the browser. Google support hinted it might be an issue with the hosting (Vercel).

### Steps to Reproduce

My app had no changes whatsoever when it suddenly, out of nowhere, stopped being indexed by Google:
![image](https://github.com/vercel/community/assets/20744582/93676e2c-1d03-48b2-8e2c-5c4665669a2d)

All tries to index it end up with the 500 error even though the app is fully working and accessible through the browser:
<img width="886" alt="CleanShot 2024-05-10 at 15 01 27@2x" src="https://github.com/vercel/community/assets/20744582/0f8acfc0-5edf-4503-ba36-55f234cc5a5d">

The `robots.txt` file allows for all crawlers:
```txt
# Allow all crawlers
User-agent: *
Allow: /
```

The app runs on Next.js version `14.0.3` with the App Router.

At this point, I'm not really sure what to do, and I'd appreciate any help 🙏

**Comment by bouia** [comment]
Ran into the same issue, have you had any luck solving this?

**Comment by jeromverschoote** [comment]
Same issue here. Am I doing something wrong?

**Reply by jeromverschoote** [reply]
Solved the issue by fixing a bug in the middleware file of the project 👍

**Comment by duckception** [comment]
In my case, I had a few bugs in my code that appeared in the console and during compilation. These bugs were not so critical cause the website worked without an issue for the users. However, it was crashing the website for Google bots, and thus, the indexing wasn't working and giving me the 500 error in the Google dashboard. Once I fixed them, everything started to work. Hope this helps anyone that encounters this issue 🫡

[a comment by an automated account is not shown]
